server/src/main/java/org/opensearch/querygroup/LabelingService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.querygroup;
+
+import org.opensearch.action.IndicesRequest;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.plugins.LabelingPlugin;
+
+import java.util.EnumMap;
+import java.util.List;
+
+/**
+ * TODO: Don't know the right package to put this class, need suggestions for maintainers on this
+ * Main class to hold multiple implementations of {@link org.opensearch.plugins.LabelingPlugin}
+ * this class will facilitate access and interactions to different implementations
+ * Usage: This class should be used as a member to orchestrate the working of your {@link LabelingPlugin}
+ */
+public class LabelingService {
+    /**
+     * Enum to define what all are currently implementing the plugin
+     */
+    public enum LabelingImplementationType {
+        QUERY_GROUP_RESOURCE_MANAGEMENT,
```

Comment:
Again if `LabelingImplementationType` class is part of core, does the plugin developer need to modify the core for adding their specific value.

Reply:
This construct is little debatable, and probably need more thought. Let me think a bit about this